ddtrace/opentelemetry/span.go

```diff
@@ -38,10 +40,14 @@ type span struct {
 func (s *span) TracerProvider() oteltrace.TracerProvider { return s.oteltracer.provider }
 
 func (s *span) SetName(name string) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
 	s.attributes[ext.SpanName] = strings.ToLower(name)
 }
 
 func (s *span) End(options ...oteltrace.SpanEndOption) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
```

Comment:
It feels weird to lock here, as there will be a second mutex locked inside `s.DD.Finish`. But I found another panic if I don't lock from the beginning of the function.

Reply:
I understand this is the recommended pattern, as it's guaranteed to unlock. We could lock after the `s.finished` conditional, but then we'll be accessing a field that can be changed concurrently too.

Reply:
That's what we do for `span.finish`, so that should be fine? https://github.com/DataDog/dd-trace-go/blob/main/ddtrace/tracer/span.go#L513-L514